core/clingutils/src/TClingUtils.cxx

```diff
@@ -3383,6 +3383,7 @@ void ROOT::TMetaUtils::GetFullyQualifiedTypeName(std::string &typenamestr,
                                                  const clang::QualType &qtype,
                                                  const cling::Interpreter &interpreter)
 {
+   cling::Interpreter::PushTransactionRAII RAII(const_cast<cling::Interpreter*>(&interpreter));
```

Comment:
I suspect we can sink this call closer the the actual deserialization that happens. Maybe it'd be worth propagating the Interpreter* argument down.

Reply:
https://gist.github.com/yamaguchi1024/1aeef3139403eda9f36b41290e761c2b

Back trace looks like this and which function do you want to sink this? from cling::utils call goes into clang, so I'm not sure if we have the best function from here.

Reply:
Is pushing a transaction 'very' cheap?  If not, it might be more efficient to do it once on the outer call rather than many time in the inner calls.

Reply:
IIRC, it is creating a small temporary object and assigns a few pointers.

Reply:
Isn't it calling IncrementalParser::commitTransaction (at the end) also?

Reply:
That should be a no-op if no decls are seen.

Reply:
Fair enough.  So last question.  What is your guess-estimate of the relative cost of:
```
{  PushTransactionRAII RAII(); ... see first set of declarations .. }
{  PushTransactionRAII RAII(); ... see second set of declarations .. }
```
vs
```
{
   PushTransactionRAII RAII(); 
   ... see first set of declarations .. 
   ... see second set of declarations ..
}
```
?

Reply:
Here is the last place which has cling::Interpreter so that we can push it to the transaction. If we go to next flame we lose this information, so I think this is the good place, what do you think?

Reply:
Looks like this is stale.

@pcanal, `PushTransactionRAII` seems to be fairly optimized, I suspect we will see no practical difference. If there is we should be able to see that in our perf monitoring once the PR is merged.